src/fpm_compiler.f90

```diff
@@ -432,6 +432,48 @@ subroutine set_preprocessor_flags (id, flags, package)
 
 end subroutine set_preprocessor_flags
 
+!> Extracts the PROJECT_VERSION_MAJOR, PROJECT_VERSION_MINOR & PROJECT_VERSION_PATCH macros.
+function get_version_macros(id, version) result(version_macros)
+    !> Compiler id.
+    integer(compiler_enum), intent(in) :: id
+
+    !> Version number of the target.
+    character(len=:), allocatable, intent(in) :: version
+
+    character(len=:), allocatable :: version_macros
+    character(len=:), allocatable :: macro_definition_symbol
+    character(:), allocatable :: version_parts(:)
+    integer :: i
+
+    !> Set macro defintion symbol on the basis of compiler used
+    select case(id)
+    case default
+        macro_definition_symbol = "-D"
+    case (id_intel_classic_windows, id_intel_llvm_windows)
+        macro_definition_symbol = "/D"
+    end select
+
+    !> Check if version macros are not allocated.
+    if (.not.allocated(version_macros)) then
+        version_macros = " "
+    end if
+
+
+    !> Extract the Major, Minor & Patch number from Version Number.
+    call split(version, version_parts, delimiters='.')
```

Comment:
There is a `version_t` which does the parsing of a version number and guarantees that all three components (major, minor, patch) are initialized correctly.

Reply:
Hi @awvwgk , Sorry for getting late on this one.

I used `version_t` for version numbers but to actually get hold of `major`, `minor` and `patch` numbers. I did this in the commit 
https://github.com/arteevraina/fpm/commit/8186856b33761f343df8dec481c5b063d7659dfb#diff-3c32b450ab98df7cbf05abbaf7c96e19c556d9de123bb1f576cafe7f7ffb667bR515 but `fortran` throws error that `num` list is private. Should I make this variable public or is there a way I can get access of minor, major and patch versions from `version_t`